### ISSUE TITLE
Containers Disappear mid upgrade due to Network issues.

### ISSUE THREAD

**Issue by Lycrom** [containrrr/watchtower#915]
**Describe the bug**

When WatchTower was updating a few of my containers my Internet went down and those containers are now gone. 

**To Reproduce**

Lose Internet connection while the containers are still finding new Images.

**Expected behavior**

If the Image download fails I would expect it to just relaunch the old image.

**Environment**

<details>
 <summary><b> Logs from running watchtower container </b></summary>

```
	/home/runner/go/pkg/mod/github.com/robfig/cron@v0.0.0-20180505203441-b41be1df6967/cron.go:199 +0x76a
created by github.com/robfig/cron.(*Cron).run
	/home/runner/go/pkg/mod/github.com/robfig/cron@v0.0.0-20180505203441-b41be1df6967/cron.go:165 +0x59
"github.com/robfig/cron.(*Cron).runWithRecovery(0xc000072050, 0xddd3c0, 0xc000013240)"
	/home/runner/go/pkg/mod/github.com/robfig/cron@v0.0.0-20180505203441-b41be1df6967/cron.go:92 +0x25
github.com/robfig/cron.FuncJob.Run(0xc000013240)
	/home/runner/work/watchtower/watchtower/cmd/root.go:264 +0xb6
github.com/containrrr/watchtower/cmd.runUpgradesOnSchedule.func1()
	/home/runner/work/watchtower/watchtower/cmd/root.go:309 +0xcc
"github.com/containrrr/watchtower/cmd.runUpdatesWithNotifications(0xc000480940, 0xc000016380)"
	/home/runner/work/watchtower/watchtower/internal/actions/update.go:73 +0x6cb
"github.com/containrrr/watchtower/internal/actions.Update(0xdf5680, 0xc00036dbc0, 0xc000480940, 0x1, 0x2540be400, 0x0, 0x0, 0x0, 0x0)"
	/home/runner/work/watchtower/watchtower/internal/actions/update.go:149 +0x191
"github.com/containrrr/watchtower/internal/actions.restartContainersInSortedOrder(0xc0005fc180, 0xe, 0x10, 0xdf5680, 0xc00036dbc0, 0xc000480940, 0x1, 0x2540be400, 0x0, 0x0)"
	/home/runner/work/watchtower/watchtower/internal/actions/update.go:183 +0xf1
"github.com/containrrr/watchtower/internal/actions.restartStaleContainer(0xc00100, 0xc000167920, 0xc00028e6e0, 0xdf5680, 0xc00036dbc0, 0xc000480940, 0x1, 0x2540be400, 0x0, 0x0, ...)"
	/home/runner/work/watchtower/watchtower/pkg/container/client.go:180 +0x86
"github.com/containrrr/watchtower/pkg/container.dockerClient.StartContainer(0xdfd900, 0xc000448580, 0x1, 0x200100, 0xc000167920, 0xc00028e6e0, 0xc000421240, 0x40, 0x0, 0x0)"
	/home/runner/work/watchtower/watchtower/pkg/container/container.go:230 +0x445
"github.com/containrrr/watchtower/pkg/container.Container.runtimeConfig(0xdf0100, 0xc000167920, 0xc00028e6e0, 0xdee280)"
	/opt/hostedtoolcache/go/1.15.10/x64/src/runtime/panic.go:969 +0x1b9
"panic(0xc144a0, 0xdcc760)"
	/home/runner/go/pkg/mod/github.com/robfig/cron@v0.0.0-20180505203441-b41be1df6967/cron.go:161 +0x9e
github.com/robfig/cron.(*Cron).runWithRecovery.func1(0xc000072050)
goroutine 8 [running]:
2021/04/21 10:27:09 cron: panic running job: assignment to entry in nil map
time="2021-04-21T10:27:06Z" level=info msg="Creating /SyncThing"
time="2021-04-21T10:27:00Z" level=info msg="Stopping /SyncThing (a73136c1ce8998cf33719d2140921062af6854604437810c5beaf06a6d3daa11) with SIGTERM"
time="2021-04-21T10:26:47Z" level=info msg="Stopping /TerrariaServerTShock (c9105c193847c4f6bf13291d9f3b8cdf064af62a74e07b4d7862c5fbfcc912c9) with SIGTERM"
time="2021-04-21T10:26:37Z" level=info msg="Stopping /Sonarr (4d4f29fb508c6fb3b268b83441806cd5c931d88a0423567eb5bd0ca34ced9384) with SIGTERM"
time="2021-04-21T10:26:24Z" level=info msg="Stopping /Red-DiscordBot (ece2390b616680e1c2d01ee12066f2854c17f707619bd64adffcab4aa8d7d78b) with SIGTERM"
time="2021-04-21T10:26:10Z" level=info msg="Stopping /TerrariaTempServer (859543c204c5766d070428ca6789078ac410ece9be2539a31ed55bb62d3a2e31) with SIGTERM"
time="2021-04-21T10:25:56Z" level=info msg="Stopping /ValheimServer (ae21f077ec8be408ac6aa0e0e3f332ba2833db517d7666ffa23f5c16f8021bf1) with SIGTERM"
time="2021-04-21T10:25:49Z" level=info msg="Stopping /Jackett (7fb52dd768bd3b78b5071c02c652d0b6ebaa9ddc2b4425a0b90789632d212238) with SIGTERM"
time="2021-04-21T10:25:47Z" level=info msg="Found new linuxserver/syncthing:latest image (sha256:95e607751a17032a1ee5386286eb6360300dda0731ed7896d7ad5bcb1c429d4b)"
time="2021-04-21T10:25:34Z" level=info msg="Found new beardedio/terraria:tshock-latest image (sha256:ba9dd75b0b8976fdb09fa5f0e52d264301cd679fc5b6ef6fe7c900435990bf82)"
time="2021-04-21T10:25:31Z" level=info msg="Found new linuxserver/sonarr:latest image (sha256:6b75773b7362e35ab5de170ad142d909a8c69a7ebbf38ba21a159bd5f1b88f65)"
time="2021-04-21T10:24:41Z" level=info msg="Found new phasecorex/red-discordbot:latest image (sha256:41be73e40aee6a45fc93db4106e0945f495bd7168da041b93bd9f940478a860e)"
time="2021-04-21T10:23:59Z" level=info msg="Found new beardedio/terraria:tshock-latest image (sha256:ba9dd75b0b8976fdb09fa5f0e52d264301cd679fc5b6ef6fe7c900435990bf82)"
time="2021-04-21T10:23:39Z" level=info msg="Found new lloesche/valheim-server:latest image (sha256:fa7c110771b6275ab3e97c5c47b1ffa2ae7ed3427a027a97971ac6945c4ec0ff)"
time="2021-04-21T10:22:59Z" level=info msg="Found new linuxserver/jackett:latest image (sha256:2de462729385befc941e014070d14c80596f74d13d03e386641c8d163c9c4bec)"
```

</details>

**Comment by simonaronsson** [comment]
Hi!

Thank you for the report! That's not how it's currently implemented.
I'll go ahead and relabel this as a feature request.

**Comment by piksel** [comment]
Actually, I don't think the issue here had anything to do with network issues at all. Or at least, not the actual error above. That one should actually be solved in `v1.3.0` as the sanity check would skip updating that container since the configuration was corrupt.
Now, why was the configuration missing `ExposedPorts`? Was it due to the network problems? Perhaps. 
In any case, watchtower would now just skip the container, instead of stopping and removing all containers before it and then crashing, leaving the rest in a removed state.

I think this should be closed, and if a need for watchtower to handle network issues in any more intricate ways a new issue with clear reproduction steps should be opened.
